File: src/mytools/vifiadmin_tools/table_entity.py
import datetime
from lib.bottle import template
from src.commons.utils.mysql_utils import init_db,query
init_db( '192.168.1.212', 'vifiadmin', 'myvifi', 'ViFi' )


# aop 注解
isAopAnnotasion = False
# spring mvc 传参注解
isWebAnnotasion = True
# package_str = "package net.eoutech.commons.entity;";
package_str = "package com.vifi.vrs.webmin.commons.entity;"
entityName = "TbCronLog"
tabName = "tbCronLog"

type_mapping = {
    "252": "String",
    "253": "String",
    "3": "Integer",
    "8": "Integer",
    "1": "Integer",
    "12": "Date"
}

@query()
def queryUser( cur ):
    sql= "SELECT * FROM %s limit 0,1" % ( tabName )
    cur.execute( sql )
    pk_show = True
    str = template("""
      ${package_str}
      import java.util.Date;
      
    % if isWebAnnotasion:
      import com.alibaba.fastjson.annotation.JSONField;
      import org.springframework.format.annotation.DateTimeFormat;
    % end

    % if isAopAnnotasion:
      import javax.persistence.Id;
      import javax.persistence.Column;
      @javax.persistence.Table( name = "${tabName}" )
    % end

    public class ${entityName} {

    % for desc in cur.description:

      % if isAopAnnotasion:
        % if pk_show:
          @Id
          <% pk_show=False %>
        % end
        @Column( name = "${desc[0]}" )
      % end
      % if isWebAnnotasion and type_mapping[str(desc[1])]=='Date':
        @JSONField( format = "yyyy-MM-dd HH:mm:ss" )
        @DateTimeFormat( pattern = "yyyy-MM-dd HH:mm:ss" )
      % end
        private ${type_mapping[str(desc[1])]} ${desc[0]};
    % end



    }
    """,dict(globals(), **vars()) )


    print( str )

    return 0


queryUser()

File: src/mytools/vifiadmin_tools/tableEntity.js
var mysql = require( '../../common/util/mysqlUtils' );
var fileUtils = require( '../../common/util/fileUtil' );

mysql.initDbInfo("vifiadmin","myvifi","192.168.1.212",3306,"ViFi");

// aop 注解
var isAopAnnotasion = false;
// spring mvc 传参注解
var isWebAnnotasion = true;
// var getPackage = "package net.eoutech.commons.entity;";
var getPackage = "package com.vifi.vrs.webmin.commons.entity;";
var entityName = "TbCDR";
var tabName = "tbCDR";


// sql 类型对象,和java对象类型映射表.
var typeMapping = {
    "252": "String",
    "253": "String",
    "3": "Integer",
    "8": "Integer",
    "1": "Integer",
    "12": "Date"
};
mysql.exec( `SELECT * FROM ${tabName} limit 0,1`, [], function ( err, rows, fields ) {
    if ( err ) throw err;

    // java 实体 属性
    var javaEntity = [
        getPackage,
        "import java.util.Date;"
    ];

    isWebAnnotasion && javaEntity.push( "import com.alibaba.fastjson.annotation.JSONField;\nimport org.springframework.format.annotation.DateTimeFormat;" );
    if ( isAopAnnotasion ) {
        javaEntity.push( "import javax.persistence.Id;\nimport javax.persistence.Column;" );
        javaEntity.push( `@javax.persistence.Table( name = "${tabName}" )` );
    }

    // entity class 
    javaEntity.push( `public class ${entityName} {` );

    var fieldInx = 0;
    for ( var field of fields ) {
        var type = typeMapping[ field.type ];
        if ( isAopAnnotasion ) {
            !(fieldInx++) && javaEntity.push( `@Id` );
            javaEntity.push( `@Column( name = "${field.name}" )` );
        }

        if ( isWebAnnotasion && type == "Date" ) {
            javaEntity.push( `@JSONField( format = "yyyy-MM-dd HH:mm:ss" )` );
            javaEntity.push( `@DateTimeFormat( pattern = "yyyy-MM-dd HH:mm:ss" )` );
        }
        javaEntity.push( `private ${type} ${field.name};\n` );

    }

    javaEntity.push( `\n\n\n\n}` );

    fileUtils.print( javaEntity.join( "\n" ) );


} );
